WrightTools/data/_data.py

```diff
@@ -2005,27 +1879,25 @@ def trim(self, channel, **kwargs):
         # call trim
         return channel.trim(neighborhood=neighborhood, **inputs)
 
-    def transpose(self, axes=None, verbose=True):
-        """Transpose the dataset.
+    def transform(self, axes, verbose=True):
+        """Transform the data.
 
         Parameters
         ----------
-        axes : None or list of int (optional)
-            The permutation to perform. If None axes are simply reversed.
-            Default is None.
-        verbose : bool (optional)
-            Toggle talkback. Default is True.
+        axes : list of strings
+            List of axes.
+        verbose : boolean (optional)
+            Toggle talkback. Default is True
         """
-        if axes is not None:
-            pass
-        else:
-            axes = range(len(self.channels[0][:].shape))[::-1]
-        self.axes = [self.axes[i] for i in axes]
-        self.attrs['axis_names'] = [axis.natural_name.encode() for axis in self.axes]
-        for channel in self.channels:
-            channel.transpose(axes=axes)
-        if verbose:
-            print('data transposed to', self.axis_names)
+        # TODO: ensure that transform does not break data
+        new = []
+        current = {a.expression: a for a in self.axes}
+        for expression in axes:
+            axis = current.get(expression, Axis(self, expression))
+            new.append(axis)
+        self.axes = new
+        self.flush()
+        self._update()
```

Comment:
swap update and flush, most likely

Reply:
I don't think it matters, actually

update only refreshes the natural names